java-checks/src/main/java/org/sonar/java/checks/DateTimeFormatterMismatchCheck.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.semantic.MethodMatchers;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.tree.Arguments;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.MemberSelectExpressionTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+@Rule(key = "S5917")
+public class DateTimeFormatterMismatchCheck extends IssuableSubscriptionVisitor {
+  private static final MethodMatchers OF_PATTERN_MATCHER = MethodMatchers.create()
+    .ofTypes("java.time.format.DateTimeFormatter")
+    .names("ofPattern")
+    .addParametersMatcher("java.lang.String")
+    .addParametersMatcher("java.lang.String", "java.util.Locale")
+    .build();
+
+  private static final MethodMatchers APPEND_VALUE_MATCHER = MethodMatchers.create()
+    .ofTypes("java.time.format.DateTimeFormatterBuilder")
+    .names("appendValue")
+    .addParametersMatcher("java.time.temporal.TemporalField", "int")
+    .build();
+
+  private static final MethodMatchers DATE_TIME_FORMATTER_BUILDER = MethodMatchers.create()
+    .ofTypes("java.time.format.DateTimeFormatterBuilder")
+    .names("toFormatter")
+    .addParametersMatcher()
+    .build();
+
+  private static final Pattern WEEK_PATTERN = Pattern.compile(".*ww{1,2}.*");
+  private static final Pattern YEAR_OF_ERA_PATTERN = Pattern.compile(".*[uy]+.*");
+
+  private static final String CHANGE_WEEK_FORMAT_MESSAGE = "Change this week format to use the week of week-based year instead.";
```

Comment:
RSPEC message in that case is `Change this year format to use ChronoField.YEAR instead`. Is there a reason why the message here is different ? I think the message from RSPEC makes more sense following the primary and secondary locations that we want to highlight